app/game/game_engine/controller.rb

```diff
@@ -1,61 +1,87 @@
 module GameEngine
   module Controller
+
+    def self.save_game_state(game_state)
+      GameEngine::Cache.save_game_state(game_state)
+    end
+
+    def self.fetch_game_state(game_data)
+      GameEngine::Cache.fetch_game_state(game_data)
+    end
+
+    def self.update_game_state_time(game_state)
+      game_state.time = Time.now
+    end
+
+    def self.output_player_data(game_state, player_id)
+      GameEngine::IO.output_player_data(game_state, player_id)
+    end
+
+    def self.selection_made?(player)
+      !player.selection.empty?
+    end
+
+    def self.gamestate_time_save_output(game_state, player_id)
+      update_game_state_time(game_state)
+      save_game_state(game_state)
+      output_player_data(game_state, player_id)
+    end
+
+    def self.phase_time_check(game_state, time_input)
+      Time.now - game_state.time >= GAME_RULES[time_input]
+    end
+
     def self.advance_game(game_data, player_id)
       # This method is responsible for switching phases inside of the GameEngine::Game
       # class when passed the ActiveRecord model of the game and the player_id for the
-      # client that's making the request. This code is not very pretty and has a lot of
-      # nested conditionals.
-      # ===========================================
-      # NEEDS TO BE REFACTORED BEFORE FINAL VERSION.
-      # ===========================================
-      game_state = GameEngine::Cache.fetch_game_state(game_data)
+      # client that's making the request.
+      game_state = fetch_game_state(game_data)
       current_time = Time.now
 
-      if game_state.phase == :setup
-        if current_time - game_state.time >= GAME_RULES[:setup_time]
+      current_player = game_state.target_player(player_id.to_i)
+      player_one = game_state.player_one
+      player_two = game_state.player_two
+
+      case game_state.phase
+      when :setup
+        if phase_time_check(game_state, :setup_time)
           game_state.phase = :move
-          game_state.time = Time.now
           GameEngine::GameResolver.deal_cards(game_state)
-          GameEngine::Cache.save_game_state(game_state)
-          GameEngine::IO.output_player_data(game_state, player_id)
+          gamestate_time_save_output(game_state, player_id)
```

Comment:
@wcstorey - This name could be more descriptive. How about "save_time_and_output"?